src/validation_workflow/docker/validation_docker.py

```diff
@@ -252,8 +250,9 @@ def run_container(self, image_ids: dict, version: str) -> Any:
         self.replacements = [(f'opensearchproject/{key}:{self.major_version_number}', value) for key, value in image_ids.items()]
 
         list(map(lambda r: self.inplace_change(self.target_yml_file, r[0], r[1]), self.replacements))
-
+        os.environ["OPENSEARCH_INITIAL_ADMIN_PASSWORD"] = get_password(str(version))
         # spin up containers
-        self.docker_compose_up = f'docker-compose -f {self.target_yml_file} up -d'
+        services = "opensearch-node1 opensearch-node2" if "opensearch-dashboards" not in self.args.projects else ""
```

Comment:
Why are we separating these instead of just spin up the entire docker compose?

Reply:
Added changes to bring up the services opensearchnode1 and opensearchnode2 only when OS is validated, bringup all the services in compose file when both OS and OSD are provided.